runners/flink/src/main/java/org/apache/beam/runners/flink/PipelineTranslationOptimizer.java

```diff
@@ -60,8 +66,10 @@ public void leaveCompositeTransform(TransformHierarchy.Node node) {}
 
   @Override
   public void visitPrimitiveTransform(TransformHierarchy.Node node) {
+    AppliedPTransform<?, ?, ?> appliedPTransform = node.toAppliedPTransform(pipeline);
```

Comment:
@tgroh Is there a better way of getting the `AppliedPTransform` from the `node` here?

Reply:
You can remove the `pipeline` field and call `getPipeline()`, which is available while traversing the pipeline if you're using `PipelineVisitor.Defaults` as the root of your visitor hierarchy. Otherwise, this is basically exactly how I'd do it.